### DISCUSSION TITLE
Is a contract created when a proposal is made?

### DISCUSSION THREAD

**Discussion by akshay-gadkari** [snapshot-labs/snapshot#3513] [Q&A]
I'm trying to understand how proposals work in snapshot. Each proposal is associated with a transaction that can be seen on etherscan, but I don't see a contract anywhere.

Does a contract generate when the proposal is made, or is there a single contract used for all the proposals that doesn't directly get linked? Are contracts used at all?

**Comment by ChaituVR** [comment]
Hi @akshay-gadkari Snapshot works off-chain. refer https://docs.snapshot.org/

**Reply by akshay-gadkari** [reply]
When I look at a proposal, for example [here](https://demo.snapshot.org/#/peachjetskilanyard.eth/proposal/0x03b3785adfd8a8c602e42ee55ca00b890fa2f6be37842a85da7f6f494a45adf1), there's a link to an ethereum block in etherscan. Does this block have any connection to the proposal?

**Reply by ChaituVR** [reply]
Yes yes, it is used to calculate voting power at that particular block https://docs.snapshot.org/faq#what-does-snapshot-mean
